libs/mock-block-dock/src/datastore/use-mock-datastore.ts

```diff
@@ -220,13 +225,47 @@ export const useMockDatastore = (
                   }
                 : undefined;
 
-            const updatedEntity: Entity = {
+            /**
+             * @todo - this is 'safe' because `getEntityRevisionFromSubgraph` returns the object from within the
+             *   `Subgraph`. If it did something like spreading and creating a new object then this wouldn't mutate
+             *   the vertex in the original graph. Are we happy with this, or should we do the messy code to refetch
+             *   it here again so we can be sure we're updating the object in the subgraph.
+             */
```

Comment:
How messy would it be?

Could it be something like:
```ts
newSubgraph.vertices = newSubgraph.vertices.map(vertex => {
  if (vertex.id !== idWeCareAbout) {
    return vertex;
  }
  return { ... }; // the updated object
});

Reply:
See
[Stop mutating react state](https://github.com/blockprotocol/blockprotocol/pull/931/commits/afd5379769edf8ef9c896a0be2a1dd0720507c5e)

It's pretty gross and depends on you knowing to look up the entity by its starting bound of the temporal metadata associated with the variable axis of the subgraph.